### ISSUE TITLE
Python API?

### ISSUE THREAD

**Issue by cnk113** [broadinstitute/CellBender#142]
Hello,

I was wondering if it was possible to get a Python API for CellBender's remove background?

Best,
Chang

**Comment by sjfleming** [comment]
Interesting question.  It would be possible, but it would certainly involve some work.

Can I ask why you would want that?

Are you thinking of using it more interactively, in a jupyter notebook or something?

**Comment by cnk113** [comment]
Yeah interactive would be great, but also to be potentially used in another pipeline would be another benefit.

**Comment by sjfleming** [comment]
You could try calling the command line tool from python, as in something like this


```python
from subprocess import run


cellbender_args = ['--input', '<FILENAME>',
                   '--output', '<NAME>',
                   '--cuda', 
                   ...]

run(['cellbender', 'remove-background'] + cellbender_args)
```

**Comment by cnk113** [comment]
Ah I think there would be quite a bottlenecking if I had to save my adata file and then reopen the output.
